### PR TITLE
Custom Validation example

### PR DESCRIPTION
**Pull request by andipaetzold**

### CODE REVIEW

File: examples/custom-validation/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+- Slug (Short text)
+
+Before we publish an entry of that content type, we want to make sure the slug has the correct format. The slug should always be prefixed with the current date. On the day of writing this text that would mean the slug has to start with `2022-05-20-`. A potential final slug could be `2022-05-20-custom-validation`.
+
+## How to prevent publishing
+
+Publishing can only ever be prevented by
+
+- not having enough permissions, or
+- the entry not being valid.
```
Comment (shikaan) [inline examples/custom-validation/README.md:22]: You can also prevent publishing via unresolved `Task`s

File: examples/custom-validation/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+To ensure our app is rendered while editing the entry, it has to be assigned to the field via the Appearance tab.
+
+## Our app
+
+> The app can be cloned locally by running `npx create-contentful-app --example custom-validation`
+
+The app itself can be built in many different ways. The approach taken in this example has the goal to be easy to extend for other use cases and more complex content types.
+
+When the Field app renders, it listens to any change (`sys`, `fields`, `metadata`) of the entry and creates an object of the full entry. Now, the custom validation logic can access every single field to determine whether the entry is valid or not. If the entry is not valid, the app calls `sdk.field.setValue('false')` which updates the field to an invalid state, and publishing is blocked. Once all issues are resolved, the app runs `sdk.field.setValue('true')`.
```
Comment (shikaan) [inline examples/custom-validation/README.md:38]: nit: maybe we can add a nicer error message here (for the screenshot at least)

**Review by shikaan — APPROVED** [review]
A couple of nits, but looks good to me

File: examples/custom-validation/src/locations/Field.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+      if ((entry.fields.slug ?? '').startsWith(`${entry.fields.date}-`)) {
+        sdk.entry.fields['slug'].getForLocale(sdk.locales.default).setInvalid(false);
+      } else {
+        sdk.entry.fields['slug'].getForLocale(sdk.locales.default).setInvalid(true);
+        newErrors.push(`Slug must start with "${entry.fields.date}-"`);
+      }
+
+      setErrors(newErrors);
+      sdk.field.setValue(newErrors.length === 0 ? 'true' : 'false');
```
Comment (colomolo) [inline examples/custom-validation/src/locations/Field.tsx:30]: Does it make sense to use boolean field for this?

File: examples/custom-validation/README.md (hunk 2 above)
Comment (andipaetzold) [reply examples/custom-validation/README.md:38]: We have an error message in the screenshot? How would you phrase that "nicer"?

**Review by colomolo — APPROVED** [review]
lgtm

**Pull request merged**